### PR TITLE
Reworked styles for chat message views

### PR DESCRIPTION
**Pull request by EgorovEI**
Styles for main chat message views were reworked to: 
- remove style inheritance
- add supporting for borders and gradiented backgrounds

ChatStyle extended by adding new separate styles for message views to keep backward compatibility.

### CODE REVIEW

**Comment by EgorovEI** [comment]
!squash

File: GliaWidgets/Sources/Extensions/UIColor+Hex.swift [hunk 1]
```diff
@@ -27,7 +27,7 @@ extension UIColor {
         self.init(red: components.R, green: components.G, blue: components.B, alpha: components.a)
     }
 
-    func toHex(alpha: Bool = false) -> String? {
+    func toHex(alpha: Bool = true) -> String? {
```
Comment (gersonnoboa) [inline GliaWidgets/Sources/Extensions/UIColor+Hex.swift:30]: Why is this needed?

File: GliaWidgets/Sources/Extensions/UIColor+Hex.swift [hunk 2]
```diff
@@ -48,3 +48,9 @@ extension UIColor {
         }
     }
 }
+
+extension CGColor {
```
Comment (gersonnoboa) [inline GliaWidgets/Sources/Extensions/UIColor+Hex.swift:52]: I don't know if this is the right file to put this. No one would figure out that this extension is declared in an `UIColor` file.

File: GliaWidgets/Sources/View/Chat/Message/Content/ChoiceCard/ChoiceCardStyle.swift [hunk 3]
```diff
@@ -92,4 +93,75 @@ public final class ChoiceCardStyle: OperatorChatMessageStyle {
 
         operatorImage.apply(configuration: configuration?.userImage)
     }
+
+    func toNewChoiceCardStyle() -> Theme.ChoiceCardStyle {
```
Comment (gersonnoboa) [inline GliaWidgets/Sources/View/Chat/Message/Content/ChoiceCard/ChoiceCardStyle.swift:97]: SwiftLint error

File: GliaWidgets/Sources/View/Chat/Message/Content/ChoiceCard/ChoiceCardView.swift [hunk 4]
```diff
@@ -40,10 +47,17 @@ final class ChoiceCardView: OperatorChatMessageView {
 
     private func contentView(for choiceCard: ChoiceCard) -> UIView {
         let containerView = UIView()
-        containerView.backgroundColor = viewStyle.backgroundColor
-        containerView.layer.cornerRadius = viewStyle.cornerRadius
-        containerView.layer.borderWidth = viewStyle.borderWidth
-        containerView.layer.borderColor = viewStyle.frameColor.cgColor
+        viewStyle.background.background.unwrap {
```
Comment (gersonnoboa) [inline GliaWidgets/Sources/View/Chat/Message/Content/ChoiceCard/ChoiceCardView.swift:50]: SwiftLint error

**Review by gersonnoboa — COMMENTED** [review]
Apologies but this pull request is way too big to be checked. So I pointed out a couple of things but I am not going to go through it again, as I am too afraid to approve it even with those things fixed. The commit message is also quite minimal for such a big commit. Don't know how adding gradient support has resulted in the deprecation of all styles. It is of paramount importance to have commit descriptions that explain things for when we are not here anymore, or for example the recent discussion where PR history would be deleted.

File: GliaWidgets/ColorType.swift [hunk 5]
```diff
@@ -16,4 +16,13 @@ public enum ColorType: Equatable {
             return false
         }
     }
+
+    var color: UIColor {
+        switch self {
+        case let .fill(color):
+            return color
+        case let .gradient(colors):
+            return UIColor(cgColor: colors.first ?? .clear)
```
[the remark this answers is not in the document]
Comment (EgorovEI) [reply GliaWidgets/ColorType.swift:25]: > dose it mean that we ignore gradient and we will think in the future how to apply gradient and rework it?

This computed property is used to transform `ColorTyle` in new style to `UIColor` in old deprecated style. So we will remove it along with deprecated styles.

Regardless that I will consider adding `applyBackground(_ background: Theme.Layer)` for `UIView`

File: GliaWidgets/Sources/Extensions/UIColor+Hex.swift (hunk 1 above)
Comment (EgorovEI) [reply GliaWidgets/Sources/Extensions/UIColor+Hex.swift:30]: We have one color with `0.5` alpha component value in default color pallet. So we need to be able transform it to hex without losing `alpha`. 
We have `UIColor.hex` computed property to convert UIColor instance to hex-string value. But it omits `alpha` component if it exists. I found this `toHex` method that was added and has never been used in Widgets. So I changed default value to `true`, because it does not affect other places.

Comment (igorkravchenko) [reply GliaWidgets/Sources/Extensions/UIColor+Hex.swift:30]: `toHex` is not very self explanatory name, because it not possible to get what kind of color hex representation it is (is it RGB/RGBA, ARGB?). If it is RGB hex value, then it has to be something like `toRGBHex`. Does it make sense?

**Review by igorkravchenko — COMMENTED** [review]
I think `toHex` method can be improved to be more self explanatory. I left [comment](https://github.com/salemove/ios-sdk-widgets/pull/722/files#r1297460992) regarding this. Also it make sense to extend PR description a bit for more information. As I understand a lot of intermediate code was added to provide backward compatibility with old styling approach and to be able to add new more straightforward one.

File: GliaWidgets/ColorType.swift (hunk 5 above)
[the remark this answers is not in the document]
Comment (EgorovEI) [reply GliaWidgets/ColorType.swift:25]: Fixed 28138aa40ffb224181edf1b751a612cf2c061f01

File: GliaWidgets/Sources/View/Chat/ChatStyle.Deprecated.swift [hunk 6]
```diff
@@ -0,0 +1,117 @@
+import UIKit
+
+extension ChatStyle {
+    /// Style of the message sent by the operator.
```
[the remark this answers is not in the document]
Comment (EgorovEI) [reply GliaWidgets/Sources/View/Chat/ChatStyle.Deprecated.swift]: 28138aa40ffb224181edf1b751a612cf2c061f01

File: GliaWidgets/Sources/Extensions/UIColor+Hex.swift (hunk 1 above)
Comment (EgorovEI) [reply GliaWidgets/Sources/Extensions/UIColor+Hex.swift]: Fixed 28138aa40ffb224181edf1b751a612cf2c061f01

File: GliaWidgets/Sources/Extensions/UIColor+Hex.swift (hunk 2 above)
Comment (EgorovEI) [reply GliaWidgets/Sources/Extensions/UIColor+Hex.swift]: Fixed 28138aa40ffb224181edf1b751a612cf2c061f01

File: GliaWidgets/Sources/View/Chat/Message/Content/ChoiceCard/ChoiceCardStyle.swift (hunk 3 above)
Comment (EgorovEI) [reply GliaWidgets/Sources/View/Chat/Message/Content/ChoiceCard/ChoiceCardStyle.swift:98]: Fixed 28138aa40ffb224181edf1b751a612cf2c061f01

File: GliaWidgets/Sources/View/Chat/Message/Content/ChoiceCard/ChoiceCardView.swift (hunk 4 above)
Comment (EgorovEI) [reply GliaWidgets/Sources/View/Chat/Message/Content/ChoiceCard/ChoiceCardView.swift]: Fixed 28138aa40ffb224181edf1b751a612cf2c061f01

**Comment by EgorovEI** [comment]
> I think `toHex` method can be improved to be more self explanatory. I left [comment](https://github.com/salemove/ios-sdk-widgets/pull/722/files#r1297460992) regarding this. Also it make sense to extend PR description a bit for more information. As I understand a lot of intermediate code was added to provide backward compatibility with old styling approach and to be able to add new more straightforward one.

Updated commit message

**Review by igorkravchenko — APPROVED** [review]

**Review by yurii-glia — APPROVED** [review]
